### ISSUE TITLE
@State property is not updated after .tap()

### ISSUE THREAD

**Issue by gedaiu** [nalexn/ViewInspector#130]
I tested the component in the app and it works fine, but the test fails because there state is somehow not updated even though the action is triggered. Any thoughts about what I am missing here?

```swift
struct FieldOptions: View {
    @State var isOpen: Bool = false
    internal let inspection = Inspection<Self>()

    var body: some View {
        VStack {
              Button("change", action: {
                  self.isOpen.toggle()
              })
          }
        .onReceive(inspection.notice) { self.inspection.visit(self, $0) }
    }
}

internal final class Inspection<V> {

    let notice = PassthroughSubject<UInt, Never>()
    var callbacks = [UInt: (V) -> Void]()

    func visit(_ view: V, _ line: UInt) {
        if let callback = callbacks.removeValue(forKey: line) {
            callback(view)
        }
    }
}

```

```swift
import Foundation
import Quick
import Nimble
import ViewInspector
import SwiftUI
@testable import MyApp

extension Inspection: InspectionEmissary { }
extension FieldOptions: Inspectable {}

class View_FieldOptionsTests: QuickSpec {
    override func spec() {
        var value: Binding<Attribute>?
        var subject: FieldOptions?

        describe("FieldOptions view") {
                beforeEach {
                    value = Binding(get: { .string("option1") }, set: { _ in })
                    subject = FieldOptions(value: value!, options: "option1,option2,option3")

                    let exp = subject!.inspection.inspect(after: 0.2) { _ in
                    }
                    ViewHosting.host(view: subject)
                    self.wait(for: [exp], timeout: 0.3)
                }

                it("renders the options when the change button is pressed") {
                    let exp = subject!.inspection.inspect { view in
                        expect(subject!.isOpen).to(equal(false))
                        try view.find(ViewType.Button.self).tap()
                        expect(subject!.isOpen).to(equal(true))
                    }

                    ViewHosting.host(view: subject)
                    self.wait(for: [exp], timeout: 0.1)
                }
            }
    }
}

```

**Comment by nalexn** [comment]
Hey @gedaiu ,

The structure of your test has to be reworked. The `beforeEach` section should not have `self.wait(for: [exp], timeout: 0.3)` or `ViewHosting.host(view: subject)`, but only the setup code.

**Comment by nalexn** [comment]
I assume you have resolved the issue. Feel free to re-open
